### ISSUE TITLE
No decoded data

### ISSUE THREAD

**Issue by Adamo198** [druckgott/ELRS-Backpack-Example-ESPNOW#2]
Hi,
Do you have any idea what am I doing wrong if I'm receiving only HEX values, without any translated telemetry? ELRS TX has connection with drone, satelites are present, I even tried to arm it, nothing changed.

```
ESPNow: len: 28, 24 58 3C 0 11 0 13 0 EA 11 2 1E C6 96 B6 A FF 25 D3 0 E 81 B0 3 E5 7 6F CC
ESPNow: len: 21, 24 58 3C 0 11 0 C 0 EA A 8 0 F4 0 A 0 0 95 5A 90 1F
ESPNow: len: 23, 24 58 3C 0 11 0 E 0 EA C 14 B5 0 64 B 0 2 1 9B 5C 6 BB 81
ESPNow: len: 19, 24 58 3C 0 11 0 A 0 EA 8 1E 0 8B 0 8B 26 1C 9C 9F
ESPNow: len: 23, 24 58 3C 0 11 0 E 0 EA C 14 B6 0 64 A 0 2 1 9B 5C 7 C9 81
ESPNow: len: 16, 24 58 3C 0 11 0 7 0 EA 5 21 4F 4B 0 97 96
ESPNow: len: 15, 24 58 3C 0 11 0 6 0 EA 4 7 0 2F FA 60
ESPNow: len: 23, 24 58 3C 0 11 0 E 0 EA C 14 B5 0 64 C 0 2 1 9B 58 7 4 81
ESPNow: len: 23, 24 58 3C 0 11 0 E 0 EA C 14 B5 0 64 C 0 2 1 9B 40 7 EC 81
```

**Comment by druckgott** [comment]
If the hex do Not fit to any thing on the crossfire Protokoll. There will be no output of text
So do you have maybe other types then the types in the crsf Code part.

Here Are the ids
https://github.com/druckgott/ELRS-Backpack-Example-ESPNOW/blob/89d189d62eaecc5fa38aad4971ee5dc67dc54bf3/lib/terseCRSF/terseCRSF.h#L72

And here for example is where it finde gps
https://github.com/druckgott/ELRS-Backpack-Example-ESPNOW/blob/89d189d62eaecc5fa38aad4971ee5dc67dc54bf3/src/crsf.cpp#L56

And if there is Not hing this an availibe id you only See the hex.
So maybe there is missing some ids in the Code. I did Not Check All.

If there is any more complete lib for crsf we can Update this in the example.

What content did you expect in the output?

Here you can finde for example the bf ids for crsf
https://github.com/betaflight/betaflight/blob/896c8ee29bffc8dde610a07bfebf05165d2cee2a/src/main/rx/crsf_protocol.h#L54

**Comment by Adamo198** [comment]
Thank you for response. I don't know why the IDs could be different in my case. I'm using Radiomaster Nomad TX with ESPNOW backpack telemetry, standard ELRS link.

I'm expecting exactly the same output as you provided on screenshot. The most interesting part are GPS parameters. This is a CRSF protocol, so ID should be 02, the same as in your case.
I will try to find if another ID will work. I tried Betaflight and INAV drones, both had 20+ parameters in telemetry, including GPS position, so it should be visible.

**Comment by druckgott** [comment]
ok than it´s strange, did you check if you have satelits. But als you descriped it should work. If you finde the solution would be cool if you can provide it here.

**Comment by Adamo198** [comment]
I solved this problem. Generally, I tried to do a port of your solution for my ESP32. I used ChatGPT (haha) to find appropriate libraries correspoding to ESP8266 and chat messed up something in this line:
`memcpy(&crsf.crsf_buf, &*(incomingData+8), sizeof(crsf.crsf_buf));`
I have also found my ESP8266 and tested it on this board, it works like it should. So I can confirm it works on ESP8266 and also ESP32.
